modin/experimental/xgboost/xgboost_ray.py

```diff
@@ -352,14 +326,22 @@ def create_actors(num_actors):
         Pair of actors list and placement group of actors.
     """
     num_cpus_per_actor = _get_cpus_per_actor(num_actors)
-    pg = _create_placement_group(num_cpus_per_actor, num_actors)
+    nodes_ips = [
+        key for key, _ in ray.cluster_resources().items() if key.startswith("node:")
+    ]
+    num_actors_per_node = num_actors // len(nodes_ips)
+    actors_ips = [ip for ip in nodes_ips for _ in range(num_actors_per_node)]
+
     actors = [
-        ModinXGBoostActor.options(
-            num_cpus=num_cpus_per_actor, placement_group=pg
-        ).remote(i, nthread=num_cpus_per_actor)
-        for i in range(num_actors)
+        (
+            node_ip.split("node:")[-1],
```

Comment:
The current logic of parts distribution assumes consistent placement of actors in container. In case of using `dict` (`{"ip": [actor_0, actor_1], ..}`) function `_assign_row_partitions_to_actors` should be significantly changed that is beyond of the scope of current problem.

Reply:
ok